test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -643,6 +644,75 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
 		})
 	})
+
+	Context("Offlined CPU API", func() {
+		BeforeEach(func() {
+			By("Saving the old performance profile")
+			initialProfile = profile.DeepCopy()
+		})
+
+		It("[disruptive] should set offline cpus after deploy PAO", func() {
+			for _, node := range workerRTNodes {
+				onlineCPUCount, err := nodes.ExecCommandOnNode([]string{"nproc", "--all"}, &node)
+				Expect(err).ToNot(HaveOccurred())
+				onlineCPUInt, err := strconv.Atoi(onlineCPUCount)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(onlineCPUInt).Should(BeNumerically(">=", 3))
+			}
+
+			// Create new performance with offlined
+			reserved := performancev2.CPUSet("0")
+			isolated := performancev2.CPUSet("1")
+			offlined := performancev2.CPUSet("2-3")
+			profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Enable UserLevelNetworking and add Devices in Profile")
+			profile.Spec.CPU = &performancev2.CPU{
+				Reserved: &reserved,
+				Isolated: &isolated,
+				Offlined: &offlined,
+			}
+			By("Updating the performance profile")
+			profiles.UpdateWithRetry(profile)
+
+			By("Applying changes in performance profile and waiting until mcp will start updating")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+
+			By("Waiting for MCP being updated")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
+			//Check offlined cpus are setting correctly
+			for _, node := range workerRTNodes {
```

Comment:
(continue the discussion here for cleanliness).

Working in local.

Failing when node reboots but should be do the check only after waiting for MCP being updated, then node should be finish the reboot and be available, right?

Reply:
yes, provided we wait for the right conditions (let's check)